src/components/Summary/GenericSummary.js

```diff
@@ -0,0 +1,100 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Card from 'components/Card';
+import ErrorMessage from 'components/ErrorMessage';
+import FormStepSummary from 'components/FormStepSummary';
+import {LayoutColumn} from 'components/Layout';
+import Loader from 'components/Loader';
+import LogoutButton from 'components/LogoutButton';
+import Price from 'components/Price';
+import SummaryConfirmation from 'components/SummaryConfirmation';
+import {SUBMISSION_ALLOWED} from 'components/constants';
+
+const GenericSummary = ({
+  title,
+  submissionAllowed,
+  summaryData = [],
+  showPaymentInformation,
+  amountToPay,
+  privacyInformation,
+  editStepText,
+  isLoading,
+  isAuthenticated,
+  errors,
+  onPrivacyCheckboxChange,
+  onSubmit,
+  onLogout,
+  onPrevPage = null,
+}) => {
+  const Wrapper = submissionAllowed === SUBMISSION_ALLOWED.yes ? 'form' : 'div';
+
+  if (isLoading) {
+    return (
+      <LayoutColumn>
+        <Loader modifiers={['centered']} />
+      </LayoutColumn>
+    );
+  }
```

Comment:
At some point we should check if [`Suspense`](https://react.dev/reference/react/Suspense) can be used